### PR TITLE
#167735055 Build Search Functionality Endpoint 

### PR DESCRIPTION
**Pull request by hameedhub**
**What does this PR do?**
  
- Have user search request record

  **Description of Task to be completed?**

  - Have user search request endpoint working

  - ` GET/  api/v1/request/search?<key>=<value>`

  **How should this be manually tested?**

  Clone the repo cd into it and RUN npm run server

 - Using postman test the above endpoint with
      key : Content-Type: value : application/json

  - To get authentication, Login or Signup and get TOKEN
      key: Authorization value: TOKEN

  **Any background context you want to provide?**

  - None

  **What are the relevant stories?**

  #167735055

  **Screenshots (if appropriate)**

 
Screen Shot 2019-09-08 at 11 18 11 AM

  **Questions:**

 - None

### CODE REVIEW

**Review by SamuelKoroh — CHANGES_REQUESTED** [review]
Nice job mate, but I was expecting the request URL to be simpler like this http://localhost:3000/api/v1/requests/?search='destination' for matching destinations,
http://localhost:3000/api/v1/requests/?search='source' for matching source,
http://localhost:3000/api/v1/requests/?search='travelDate' etc.

**Comment by hameedhub** [comment]
> Nice job mate, but I was expecting the request URL to be simpler like this http://localhost:3000/api/v1/requests/?search='destination' for matching destinations,
> http://localhost:3000/api/v1/requests/?search='source' for matching source,
> http://localhost:3000/api/v1/requests/?search='travelDate' etc.

Thank you very much for the feedback @SamuelKoroh, I was told by TTL to pass the query this way i.e `api/requests/search?<key>=<value>`, i would make sure i call her attention to your review.. Thank you once again..

**Review by SamuelKoroh — APPROVED** [review]

File: src/routes/request.js [hunk 1]
```diff
@@ -10,12 +10,16 @@ const {
...
+  returnRequest,
+  search
 } = RequestController;
 
 router.post('/multi-city', [auth, validator(requestSchema)], multiCityRequest);
 router.post('/one-way', [auth, validator(requestSchema)], oneWay);
 router.patch('/:requestId', [auth, validator(requestSchema)], update);
 router.get('/', auth, getRequests);
+router.post('/return', [auth, validator(requestSchema)], returnRequest);
+router.get('/search?', auth, search);
```
Comment (hameedhub) [inline src/routes/request.js]: remove search

Comment (Annmary12) [inline src/routes/request.js]: I don't feel it is okay to leave our route this way `router.get('/search?', auth, search)`, change line `21` : `router.get('/', auth, getRequests);` to  `router.get('/userRequests', auth, getUserRequests);`
then yours will be this way below

```suggestion
router.get('/', auth, getRequests);
```

Please ensure that the route you just changed is working properly

File: src/services/authService.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ export default class AuthService {
     const payload = Helper.pickFields(user, ['id', 'roleId']);
     const token = await Helper.genToken(payload);
 
-    return { token, ...Helper.omitFields(user, ['password']) };
+    return { token, ...Helper.omitFields(user, ['password', 'createdat', 'updatedat']) };
```
Comment (Annmary12) [inline src/services/authService.js:118]: I thought @SamuelKoroh fixed this {'createdat', 'updatedat'}, why do we still have it?

File: src/tests/request.test.js [hunk 3]
```diff
@@ -263,4 +264,133 @@ describe('/api/v1/requests', () => {
       expect(JSON.parse(res.text).error).to.equal('You\'ve not make any requests');
     });
   });
+
+  describe('POST / ReturnTrip', () => {
```
Comment (Annmary12) [inline src/tests/request.test.js:333]: I thought I have merged your return trip task, why is the changes still here?

**Comment by Annmary12** [comment]
Also resolve conflict on your PR

**Review by hameedhub — COMMENTED** [review]
> Also resolve conflict on your PR

Noted

File: src/routes/request.js (hunk 1 above)
Comment (hameedhub) [reply src/routes/request.js]: Noted

File: src/tests/request.test.js (hunk 3 above)
Comment (hameedhub) [reply src/tests/request.test.js:333]: I checkout from the branch while creating this current branch

**Pull request merged**